Adding pending test for #269

romanowski opened the pull request:
Zinc is undercompiling in this case. Reported as #269 

For certain reasons this time I have to paste the below disclaimer:
THIS PROGRAM IS SUBJECT TO THE TERMS OF THE BSD 3-CLAUSE LICENSE.

THE FOLLOWING DISCLAIMER APPLIES TO ALL SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE:
THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR
ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.
ONLY THE SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE, IF ANY, THAT ARE ATTACHED TO (OR OTHERWISE ACCOMPANY) THIS SUBMISSION (AND ORDINARY
COURSE CONTRIBUTIONS OF FUTURES PATCHES THERETO) ARE TO BE CONSIDERED A CONTRIBUTION. NO OTHER SOFTWARE CODE OR MATERIALS ARE A CONTRIBUTION.

romanowski committed:
Add 'fuzzy type` pending test for #269

dwijnand commented on zinc/src/sbt-test/source-dependencies/fuzzyTypes/pending:
```diff
@@ -0,0 +1,6 @@
+> compile
+> compile
```
why compile twice?

romanowski replied:
yeah, leftover from testing :)

dwijnand commented on zinc/src/sbt-test/source-dependencies/fuzzyTypes/pending:
```diff
@@ -0,0 +1,5 @@
+$ copy-file changed/FactoryProvider.scala src/main/scala/foo/FactoryProvider.scala
+-> compile
+> clean
+-> compile
```
If Zinc is under-compiling shouldn't this be a success after a clean?

jvican committed:
Rename directory to stay consistent

Rename the `fuzzyTypes` directory to the more consistent `fuzzy-types`
since the rest of the directories use this style.

jvican commented:
Will merge this as soon as CI passes because I'll need to rebase for #272.

eed3si9n commented:
Is "fuzzy type" an actual term in type theory here?
The term fuzzy reminds me only of [fuzzy logic](https://en.wikipedia.org/wiki/Fuzzy_logic) that was really popular among the Japanese appliance makers in the 90s.

jvican commented:
@eed3si9n Yeah, actually it sounds weird to me too. Do you think `scalac-type-expansion` would be more correct?

eed3si9n commented:
Not sure what was the problematic part, but I'm guessing it'd be something more boring like type-projection, parameterized-type, type-reference, as-seen-from-c etc.

jvican commented:
The root of the problem is scalac doing type expansions that we don't track, so I would go for `expanded-type-projection`.

eed3si9n commented:
Sounds good to me.

jvican committed:
Rename `fuzzy-types` to `expanded-type-projection`

Just to be more specific.

jvican reviewed (approved):
LGTM, thanks @romanowski.
